distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc

```diff
@@ -0,0 +1,152 @@
+||cDEF           |                   |cDEF           Eventing                   |    +--------+|
+||   Catalog     |                   +------------------------------------------+              |
+||   Plugins     |                   |cDEF           Resources                  |              |
+|\---------------+-------------------+------------------------------------------/              |
+|                |c369               |                                                         |
+|                | Catalog Provider  |              cCCC                                       |
+|                +-------------------+                                                         |
+|                |cDEF               |                                                         |
+|                | Storage Provider  |                                                         |
+|                \-------------------/                                                         |
+|                          |                                                                   |
+|                          v                                                                   |
+|                  +-=-------------+                                                           |
+|                  |               |                                                           |
+|                  |{s}Data Store  |                                                           |
+|                  +---------------+                                                           |
++----------------------------------------------------------------------------------------------+
+....
+
+==== Implement a Source Interface
+
+There are three types of sources that can be created.
+All of these types of sources can perform a query operation.
+Operating on queries is the foundation for all sources.
+All of these sources must also be able to return their availability and the list of content types currently stored in their back-end data stores.
+
+Catalog Provider:: `ddf.catalog.source.CatalogProvider` _is used to communicate with back-end storage and allows for Query and Create/Update/Delete operations._
+Federated Source:: `ddf.catalog.source.FederatedSource` _is used to communicate with remote systems and only allows query operations._
+Connected Source:: `ddf.catalog.source.ConnectedSource` _is similar to a Federated Source with the following exceptions:_
+* _Queried on all local queries_
+* _`SiteName` is hidden (masked with the ${branding} sourceId) in query results_
+* _`SiteService` does not show this Source's information separate from ${branding}'s._
+Catalog Store:: `catalog.store.interface` is used to
+
+The procedure for implementing any of the source types follows a similar format:
+
+. Create a new class that implements the specified Source interface and `ConfiguredService`.
+. Implement the required methods.
```

Comment:
Create a new class that implements the specified Source interface, the `ConfiguredService` and the required methods.